# Clase 18

In [1]:
import math

## Solución del examen 1

<img src="e01-Q1.png" align="middle" width="600">

No se pueden cumplir estas dos condiciones simultanéamente:
* La tercera carga debe ser colineal a las dos cargas y encontrarse fuera de la línea que las une, y
* la tercera carga debe ser equidistante a las otras dos.

(Excepto, tal vez, en infinito.)

$$ \circledast $$

<img src="e01-Q2.png" align="middle" width="600">

No existe modificación ya que la fuerza
$$ \vec{F}_{12}=k_e\dfrac{q_1q_2}{r_{12}^2}\hat{r}_{12} $$
sólo depende de la magnitud de las cargas y la distancia que las separa. Entonces, aunque pusiéramos otras cargas, no modificaría esta interacción pero la fuerza neta sobre una de las cargas sí. Si dejamos transcurrir el tiempo y una de las cargas se mueve por interacciones con las otras tal vez cambie la distancia por lo que podría cambiar la magnitud de la fuerza $\vec{F}_{12}$.

$$ \circledast $$

<img src="e01-Q3.png" align="middle" width="600">

No, porque seguirá la trayectoria de los vectores tangentes a la línea de campo. Sólo podría seguir la línea de campo si se tratara de un campo uniforme.

$$ \circledast $$

<img src="e01-P1.png" align="middle" width="600">

Se hizo en clase.

<img src="e01-P2.png" align="middle" width="600">

(b)

$$ W=\int_{\vec{r}_0}^{\vec{r}_1}\vec{F}_\text{externa}\cdot d\vec{r} $$

\begin{align}
W&=\int_\infty^{d}(-\vec{F})\cdot d\vec{r}=-\int_\infty^{d}\left(k_e\dfrac{q_1q_2}{r^2}\hat{r}\right)\cdot d\vec{r}\\
&=-k_eq_1q_2\int_\infty^{d}\dfrac{dr}{r^2}\\
&=\left.k_e q_1q_2\dfrac{1}{r}\right|_\infty^d\\
&=k_e q_1q_2\left(\dfrac{1}{d}-\lim_{r\to\infty}\dfrac{1}{r}\right)\\
&=k_e\dfrac{q_1q_2}{d}<0
\end{align}

$$ d=\sqrt{(x_1-x_2)^2+(y_1-y_2)^2} $$

In [2]:
d = math.sqrt(((3.5)-(-2.))**2+((0.5)-(1.5))**2)
d

5.5901699437494745

$$ d\approx5.59 \text{cm} $$

In [3]:
kₑ = 9e9
q_1 = -3e-6
q_2 = 4e-6
kₑ*q_1*q_2/(d*1e-2)

-1.931962732559818

$$ W=-1.93\ \text{J} $$

Parece que yo estaba equivocado y el profesor estaba en lo correcto: **el trabajo para acercar dos cargas de signo opuestos es negativo.** Si quieren entender esto mejor sugiero revisar los siguientes links:
* https://physics.stackexchange.com/questions/482111/why-there-is-a-negative-sign-in-the-formula-of-calculating-work-done-in-electros
* https://physics.stackexchange.com/questions/242170/electric-potential-of-point-charge-sign-problem
* https://physics.stackexchange.com/questions/481126/why-the-work-done-is-positive-when-bringing-2-opposite-charges-together
* https://www.physicsforums.com/threads/why-the-work-done-is-negative-when-bringing-2-opposite-charges-together.972183/

<img src="e01-P3.png" align="middle" width="600">

<img src="e01-F1.png" align="middle" width="600">

$$ d\vec{E}_+=k_e\dfrac{dq}{R^2}(\cos\theta\hat{x}-\sin\theta\hat{y}) $$

$$ d\vec{E}_-=k_e\dfrac{dq}{R^2}(-\cos\theta\hat{x}+\sin\theta\hat{y}) $$

$$ d\vec{E}=k_e\dfrac{dq}{R^2}(-2\sin\theta\hat{y})=k_e\dfrac{\lambda\,ds}{R^2}(-2\sin\theta\hat{y})=-\hat{y}\,2k_e\dfrac{\lambda\,d\theta}{R}\sin\theta $$

$$ \vec{E}=-\,2k_e\dfrac{\lambda}{R}\hat{y} $$

<img src="e01-E1.png" align="middle" width="600">

$$ \vec{E}(z)=-\,2k_e\dfrac{\lambda R^2}{(R^2+z^2)^{3/2}}\hat{y}=-\dfrac{Q R}{\pi^2\epsilon_0(R^2+z^2)^{3/2}}\hat{y} $$

La fuerza sobre un pequeño dipolo en un campo no uniforme está dado por

$$ \vec{F}=(\vec{p}\cdot \nabla)\vec{E}=\left(p_x\dfrac{\partial \phantom{x}}{\partial x}+p_y\dfrac{\partial\phantom{y}}{\partial y}+p_z\dfrac{\partial\phantom{ z}}{\partial z}\right)\vec{E} $$

$$ \vec{p}=\pm p\hat{z} $$

$$ \vec{F}=\pm p\dfrac{\partial\phantom{ z}}{\partial z}\vec{E}=\pm\dfrac{3QpRz}{\pi^2\epsilon_0(R^2+z^2)^{5/2}}\hat{y} $$

<img src="e01-E2.png" align="middle" width="600">

<img src="cargas_positivas.png" align="middle" width="300">

### Ángulo entre tangentes del campo eléctrico de dos cargas positivas

Supongamos que el campo eléctrico del sistema es el de $2q$, pero en el punto $P(-a,0)$ y que la separación entre la dos cargas es $2a$. Podemos tomar coordenadas polares:

$$ \vec{E}(r,\theta)=\dfrac{2k_eq}{r^2}\left[\left(1+\dfrac{1+z\cos\theta}{[1+2z\cos\theta+z^2]^{3/2}}\right)\hat{r}-\dfrac{z\sin\theta}{[1+2z\cos\theta+z^2]^{3/2}}\hat{\theta}\right] $$

con $z=a/r$, donde $r$ es la distancia lejana en la que mediremos el nuevo ángulo.

In [4]:
from sympy import *

Definimos nuestras variables:

In [5]:
z, θ = symbols('z θ', real=True)

In [6]:
θ1, θ2 = symbols('θ1 θ2', real=True)

In [7]:
ϕ1 = symbols('ϕ1', real=True)

In [8]:
ϕ2 = symbols('ϕ2', real=True)

Como $r\gg a$, haremos las aproximaciones $z\ll1$ y $\theta\ll1$ en expansiones de Taylor.

Sea $f(z)=(1+z)^{-n}=1-nz+\dfrac{n(n+1)}{2}z^2+\mathcal{O}(z^3)$

Veamos que le pasa a la expansión de Taylor de

$$ [1+2z\cos\theta+z^2]^{-3/2} $$

Entonces

In [9]:
n_exp_den_gral = Rational(3,2)

In [10]:
den_gral = 1-n_exp_den_gral*(2*z*cos(θ)+z**2)+Rational(n_exp_den_gral*(n_exp_den_gral+1),2)*(2*z*cos(θ)+z**2)**2
expand(den_gral)

15*z**4/8 + 15*z**3*cos(θ)/2 + 15*z**2*cos(θ)**2/2 - 3*z**2/2 - 3*z*cos(θ) + 1

$$ [1+2z\cos\theta+z^2]^{-3/2}=\displaystyle \frac{15 z^{4}}{8} + \frac{15 z^{3} \cos{\left(θ \right)}}{2} + \frac{15 z^{2} \cos^{2}{\left(θ \right)}}{2} - \frac{3 z^{2}}{2} - 3 z \cos{\left(θ \right)} + 1+\mathcal{O}(z^5) $$

Veamos que le pasa a la expansión de Taylor de

$$ 1+\dfrac{1+z\cos\theta}{[1+2z\cos\theta+z^2]^{3/2}} $$

In [11]:
1+expand((1+z*cos(θ))*expand(den_gral))

15*z**5*cos(θ)/8 + 15*z**4*cos(θ)**2/2 + 15*z**4/8 + 15*z**3*cos(θ)**3/2 + 6*z**3*cos(θ) + 9*z**2*cos(θ)**2/2 - 3*z**2/2 - 2*z*cos(θ) + 2

In [12]:
E_r = expand((1+expand((1+z*cos(θ))*expand(den_gral))).subs(cos(θ), 1-Rational(1,2)*θ**2))
E_r

-15*z**5*θ**2/16 + 15*z**5/8 + 15*z**4*θ**4/8 - 15*z**4*θ**2/2 + 75*z**4/8 - 15*z**3*θ**6/16 + 45*z**3*θ**4/8 - 57*z**3*θ**2/4 + 27*z**3/2 + 9*z**2*θ**4/8 - 9*z**2*θ**2/2 + 3*z**2 + z*θ**2 - 2*z + 2

y que le la expansión de Taylor de

$$ -\dfrac{z\sin\theta}{[1+2z\cos\theta+z^2]^{3/2}} $$

es

In [13]:
expand(-z*sin(θ)*expand(den_gral))

-15*z**5*sin(θ)/8 - 15*z**4*sin(θ)*cos(θ)/2 - 15*z**3*sin(θ)*cos(θ)**2/2 + 3*z**3*sin(θ)/2 + 3*z**2*sin(θ)*cos(θ) - z*sin(θ)

In [14]:
E_θ = expand((expand(-z*sin(θ)*expand(den_gral))).subs([(sin(θ), θ), (cos(θ), 1-Rational(1,2)*θ**2)]))
E_θ

-15*z**5*θ/8 + 15*z**4*θ**3/4 - 15*z**4*θ/2 - 15*z**3*θ**5/8 + 15*z**3*θ**3/2 - 6*z**3*θ - 3*z**2*θ**3/2 + 3*z**2*θ - z*θ

Entonces el vector $\vec{E}$ en componentes es

In [15]:
E = Matrix([[E_r], [E_θ]])
E

Matrix([
[-15*z**5*θ**2/16 + 15*z**5/8 + 15*z**4*θ**4/8 - 15*z**4*θ**2/2 + 75*z**4/8 - 15*z**3*θ**6/16 + 45*z**3*θ**4/8 - 57*z**3*θ**2/4 + 27*z**3/2 + 9*z**2*θ**4/8 - 9*z**2*θ**2/2 + 3*z**2 + z*θ**2 - 2*z + 2],
[                                                                             -15*z**5*θ/8 + 15*z**4*θ**3/4 - 15*z**4*θ/2 - 15*z**3*θ**5/8 + 15*z**3*θ**3/2 - 6*z**3*θ - 3*z**2*θ**3/2 + 3*z**2*θ - z*θ]])

$$ \circledast $$

In [16]:
termino_expansion_taylor_norma = expand((Transpose(E)*E)[0])/4
termino_expansion_taylor_norma

225*z**10*θ**4/1024 + 225*z**10/256 - 225*z**9*θ**6/256 + 225*z**9*θ**4/128 - 1125*z**9*θ**2/256 + 1125*z**9/128 + 675*z**8*θ**8/512 - 675*z**8*θ**6/128 + 1755*z**8*θ**4/128 - 1125*z**8*θ**2/32 + 8865*z**8/256 - 225*z**7*θ**10/256 + 675*z**7*θ**8/128 - 585*z**7*θ**6/32 + 1845*z**7*θ**4/32 - 6615*z**7*θ**2/64 + 2115*z**7/32 + 225*z**6*θ**12/1024 - 225*z**6*θ**10/128 + 2205*z**6*θ**8/256 - 1215*z**6*θ**6/32 + 13053*z**6*θ**4/128 - 4095*z**6*θ**2/32 + 231*z**6/4 - 135*z**5*θ**10/256 + 855*z**5*θ**8/128 - 1893*z**5*θ**6/64 + 1803*z**5*θ**4/32 - 183*z**5*θ**2/4 + 51*z**5/4 - 39*z**4*θ**8/256 + 87*z**4*θ**6/32 - 81*z**4*θ**4/8 + 12*z**4*θ**2 - 15*z**4/8 - 3*z**3*θ**6/8 + 3*z**3*θ**4 - 39*z**3*θ**2/4 + 21*z**3/2 + 11*z**2*θ**4/8 - 21*z**2*θ**2/4 + 4*z**2 + z*θ**2 - 2*z + 1

In [17]:
termino_expansion_taylor_norma_corte = (termino_expansion_taylor_norma.coeff(z))*z + (termino_expansion_taylor_norma.coeff(z**2))*z**2
termino_expansion_taylor_norma_corte

z**2*(11*θ**4/8 - 21*θ**2/4 + 4) + z*(θ**2 - 2)

In [18]:
n_exp_den_norma = Rational(1,2)

In [19]:
expansion_taylor_norma = Rational(1,2)*expand(1-n_exp_den_norma*termino_expansion_taylor_norma_corte+Rational(n_exp_den_norma*(n_exp_den_norma+1),2)*termino_expansion_taylor_norma_corte**2)
expansion_taylor_norma

363*z**4*θ**8/1024 - 693*z**4*θ**6/256 + 1851*z**4*θ**4/256 - 63*z**4*θ**2/8 + 3*z**4 + 33*z**3*θ**6/64 - 3*z**3*θ**4 + 87*z**3*θ**2/16 - 3*z**3 - 5*z**2*θ**4/32 + 9*z**2*θ**2/16 - z**2/4 - z*θ**2/4 + z/2 + 1/2

In [20]:
expansion_taylor_norma_corte = Rational(1,2) + (expansion_taylor_norma.coeff(z))*z + (expansion_taylor_norma.coeff(z**2))*z**2
expansion_taylor_norma_corte

z**2*(-5*θ**4/32 + 9*θ**2/16 - 1/4) + z*(1/2 - θ**2/4) + 1/2

In [21]:
norma_E_1 = expansion_taylor_norma_corte.subs(θ, θ1)
norma_E_2 = expansion_taylor_norma_corte.subs(θ, θ2)

In [22]:
E_1 = E.subs(θ, θ1)
E_2 = E.subs(θ, θ2)
prod_punto_12 = expand((Transpose(E_1)*E_2)[0])
prod_punto_12

225*z**10*θ1**2*θ2**2/256 - 225*z**10*θ1**2/128 + 225*z**10*θ1*θ2/64 - 225*z**10*θ2**2/128 + 225*z**10/64 - 225*z**9*θ1**4*θ2**2/128 + 225*z**9*θ1**4/64 - 225*z**9*θ1**3*θ2/32 - 225*z**9*θ1**2*θ2**4/128 + 225*z**9*θ1**2*θ2**2/16 - 2925*z**9*θ1**2/128 - 225*z**9*θ1*θ2**3/32 + 225*z**9*θ1*θ2/8 + 225*z**9*θ2**4/64 - 2925*z**9*θ2**2/128 + 1125*z**9/32 + 225*z**8*θ1**6*θ2**2/256 - 225*z**8*θ1**6/128 + 225*z**8*θ1**5*θ2/64 + 225*z**8*θ1**4*θ2**4/64 - 2475*z**8*θ1**4*θ2**2/128 + 225*z**8*θ1**4/8 + 225*z**8*θ1**3*θ2**3/16 - 675*z**8*θ1**3*θ2/16 + 225*z**8*θ1**2*θ2**6/256 - 2475*z**8*θ1**2*θ2**4/128 + 2655*z**8*θ1**2*θ2**2/32 - 1755*z**8*θ1**2/16 + 225*z**8*θ1*θ2**5/64 - 675*z**8*θ1*θ2**3/16 + 315*z**8*θ1*θ2/4 - 225*z**8*θ2**6/128 + 225*z**8*θ2**4/8 - 1755*z**8*θ2**2/16 + 8865*z**8/64 - 225*z**7*θ1**6*θ2**4/128 + 225*z**7*θ1**6*θ2**2/32 - 1125*z**7*θ1**6/128 - 225*z**7*θ1**5*θ2**3/32 + 225*z**7*θ1**5*θ2/16 - 225*z**7*θ1**4*θ2**6/128 + 675*z**7*θ1**4*θ2**4/32 - 8955*z**7*θ1**4*θ2**2/128 + 2565*z

In [23]:
prod_punto_12_corte = 4 + (prod_punto_12.coeff(z))*z + (prod_punto_12.coeff(z**2))*z**2
prod_punto_12_corte

z**2*(9*θ1**4/4 + θ1**2*θ2**2 - 11*θ1**2 + θ1*θ2 + 9*θ2**4/4 - 11*θ2**2 + 16) + z*(2*θ1**2 + 2*θ2**2 - 8) + 4

In [24]:
cos_θp = expand(prod_punto_12_corte*norma_E_1*norma_E_2)
cos_θp

225*z**6*θ1**8*θ2**4/4096 - 405*z**6*θ1**8*θ2**2/2048 + 45*z**6*θ1**8/512 + 25*z**6*θ1**6*θ2**6/1024 - 1135*z**6*θ1**6*θ2**4/2048 + 1759*z**6*θ1**6*θ2**2/1024 - 191*z**6*θ1**6/256 + 25*z**6*θ1**5*θ2**5/1024 - 45*z**6*θ1**5*θ2**3/512 + 5*z**6*θ1**5*θ2/128 + 225*z**6*θ1**4*θ2**8/4096 - 1135*z**6*θ1**4*θ2**6/2048 + 721*z**6*θ1**4*θ2**4/256 - 739*z**6*θ1**4*θ2**2/128 + 37*z**6*θ1**4/16 - 45*z**6*θ1**3*θ2**5/512 + 81*z**6*θ1**3*θ2**3/256 - 9*z**6*θ1**3*θ2/64 - 405*z**6*θ1**2*θ2**8/2048 + 1759*z**6*θ1**2*θ2**6/1024 - 739*z**6*θ1**2*θ2**4/128 + 263*z**6*θ1**2*θ2**2/32 - 47*z**6*θ1**2/16 + 5*z**6*θ1*θ2**5/128 - 9*z**6*θ1*θ2**3/64 + z**6*θ1*θ2/16 + 45*z**6*θ2**8/512 - 191*z**6*θ2**6/256 + 37*z**6*θ2**4/16 - 47*z**6*θ2**2/16 + z**6 + 45*z**5*θ1**8*θ2**2/512 - 45*z**5*θ1**8/256 + 45*z**5*θ1**6*θ2**4/256 - 337*z**5*θ1**6*θ2**2/256 + 219*z**5*θ1**6/128 + 5*z**5*θ1**5*θ2**3/128 - 5*z**5*θ1**5*θ2/64 + 45*z**5*θ1**4*θ2**6/256 - 261*z**5*θ1**4*θ2**4/128 + 455*z**5*θ1**4*θ2**2/64 - 99*z**5*θ1**4/16 + 5*

In [25]:
prod_punto_12_corte = 1 + (cos_θp.coeff(z))*z + (cos_θp.coeff(z**2))*z**2
expand(prod_punto_12_corte)

-z**2*θ1**2/8 + z**2*θ1*θ2/4 - z**2*θ2**2/8 + 1

Lo que hicimos fue quedarnos con los términos cuadráticos de las expansiones de Taylor del cociente del producto punto y el producto de las normas de los vectores. Entonces

$$ \cos\theta'=\dfrac{\vec{E}_1(r,\theta_1)\cdot\vec{E}_2(r,\theta_2)}{E_1(r,\theta_1)E_2(r,\theta_2)}\approx1-\dfrac{z^2}{2}\left(\dfrac{\theta_2-\theta_1}{2}\right)^2 $$

$$ 1-\dfrac{\theta'^2}{2}\approx1-\dfrac{z^2}{2}\left(\dfrac{\theta_2-\theta_1}{2}\right)^2 $$

$$ \theta'_2-\theta'_1\equiv\theta'\approx z\,\dfrac{\theta_2-\theta_1}{2}=\dfrac{a}{r}\,\dfrac{\theta}{2} $$

$$ \circledast $$